mvpoly/src/prime.rs

```diff
@@ -439,6 +440,126 @@ impl<F: PrimeField, const N: usize, const D: usize> Dense<F, N, D> {
             });
         result
     }
+
+    /// Compute the cross-terms as described in [Behind Nova: cross-terms
+    /// computation for high degree
+    /// gates](https://hackmd.io/@dannywillems/Syo5MBq90)
+    ///
+    /// The polynomial must not necessarily be homogeneous. For this reason, the
+    /// values `u1` and `u2` represents the extra variable that is used to make
+    /// the polynomial homogeneous.
+    ///
+    /// The homogeneous degree is supposed to be the one defined by the type of
+    /// the polynomial, i.e. `D`.
+    ///
+    /// The output is a map of `D - 1` values that represents the cross-terms
+    /// for each power of `r`.
+    // IMPROVEME: Dummy implementation, a cache can be used to save the
+    // previously computed multiplications and powers.
+    // Maybe using a symbolic form could speed up the computation.
+    pub fn compute_cross_terms(
+        &self,
+        eval1: &[F; N],
+        eval2: &[F; N],
+        u1: F,
+        u2: F,
+    ) -> HashMap<usize, F> {
```

Comment:
Maybe this is a dumb question, but I don't understand why we use a HashMap here instead of just a Vector.

Reply:
Just noticed this is still draft.  Sorry.  Feel free to resolve.